tests/lib/rules/no-array-prototype-extensions.js

```diff
@@ -481,10 +481,27 @@ import { get as g } from 'dummy';
       errors: [{ messageId: 'main', type: 'CallExpression' }],
     },
     {
+      // When unexpected number of arguments are passed, auto-fixer will not run
       code: 'something.mapBy()',
       output: null,
       errors: [{ messageId: 'main', type: 'CallExpression' }],
     },
+    {
+      // When unexpected number of arguments are passed, auto-fixer will not run
+      code: 'something.mapBy(1, 2)',
+      output: null,
+      errors: [{ messageId: 'main', type: 'CallExpression' }],
+    },
+    {
+      code: "something.mapBy('abc')",
+      output: "something.map(item => item['abc'])",
```

Comment:
Can we use `item.abc` in this case? There's a function `isValidJSVariableName` in another rule that you can extract to a helper file for us to determine if it's safe to call with a dot. For example, `item['foo-bar']` still needs the brackets.

Reply:
I have used bracket notation to handle both static and dynamic values.

Reply:
What do you mean static and dynamic values?

Reply:
If you mean like a literal value like string vs. a variable, then we should detect that and only use brackets when necessary:

```js
item.abc
item['foo-bar']
item[123]
item[foo()]
item[foo]
```

Reply:
Also, let's be consistent about using `get`. Either use it in all the relevant autofixers or none of them?

Relevant discussion: https://github.com/ember-cli/eslint-plugin-ember/pull/1610#discussion_r989369114

Reply:
> If you mean like a literal value like string vs. a variable, then we should detect that and only use brackets when necessary:
> 
> ```js
> item.abc
> item['foo-bar']
> item[123]
> item[foo()]
> item[foo]
> ```

IMHO, it should be outside the scope of this specific lint rule. Some other lint rule should take care of removing the bracket way of accessing if it is not required so that it would be in alignment with the `separation of concerns` design principle.

Reply:
> Also, let's be consistent about using `get`. Either use it in all the relevant autofixers or none of them?
> 
> Relevant discussion: [#1610 (comment)](https://github.com/ember-cli/eslint-plugin-ember/pull/1610#discussion_r989369114)

It's my bad. switched to get now.